compute/main.go

```diff
@@ -0,0 +1,596 @@
+	return resp.PublicIPAddress, err
+}
+
+func deletePublicIP(ctx context.Context, connection *armcore.Connection, publicIPName string) (*http.Response, error) {
+	publicIPAddressClient := armnetwork.NewPublicIPAddressesClient(connection, subscriptionId)
+
+	pollerResponse, err := publicIPAddressClient.BeginDelete(ctx, resourceGroupName, publicIPName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return resp, err
+}
+
+func createNetWorkInterface(ctx context.Context, connection *armcore.Connection, subnetID *string, publicIPID *string, networkSecurityGroupID *string) (*armnetwork.NetworkInterface, error) {
+	nicClient := armnetwork.NewNetworkInterfacesClient(connection, subscriptionId)
+
+	parameters := armnetwork.NetworkInterface{
+		Resource: armnetwork.Resource{
+			Location: to.StringPtr(location),
+		},
+		Properties: &armnetwork.NetworkInterfacePropertiesFormat{
+			//NetworkSecurityGroup:
+			IPConfigurations: []*armnetwork.NetworkInterfaceIPConfiguration{
+				{
+					Name: to.StringPtr("ipConfig"),
+					Properties: &armnetwork.NetworkInterfaceIPConfigurationPropertiesFormat{
+						PrivateIPAllocationMethod: armnetwork.IPAllocationMethodDynamic.ToPtr(),
+						Subnet: &armnetwork.Subnet{
+							SubResource: armnetwork.SubResource{
+								ID: subnetID,
+							},
+						},
+						PublicIPAddress: &armnetwork.PublicIPAddress{
+							Resource: armnetwork.Resource{
+								ID: publicIPID,
+							},
+						},
+					},
+				},
+			},
+			NetworkSecurityGroup: &armnetwork.NetworkSecurityGroup{
+				Resource: armnetwork.Resource{
+					ID: networkSecurityGroupID,
+				},
+			},
+		},
+	}
+
+	pollerResponse, err := nicClient.BeginCreateOrUpdate(ctx, resourceGroupName, nicName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.NetworkInterface, err
+}
+
+func deleteNetWorkInterface(ctx context.Context, connection *armcore.Connection) (*http.Response, error) {
+	nicClient := armnetwork.NewNetworkInterfacesClient(connection, subscriptionId)
+
+	pollerResponse, err := nicClient.BeginDelete(ctx, resourceGroupName, nicName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, err
+}
+
+func createVirtualMachine(ctx context.Context, connection *armcore.Connection, networkInterfaceID string) (*armcompute.VirtualMachine, error) {
+	vmClient := armcompute.NewVirtualMachinesClient(connection, subscriptionId)
+
+	//require ssh key for authentication on linux
+	//sshPublicKeyPath := "/home/user/.ssh/id_rsa.pub"
+	//var sshBytes []byte
+	//_,err := os.Stat(sshPublicKeyPath)
+	//if err == nil {
+	//	sshBytes,err = ioutil.ReadFile(sshPublicKeyPath)
+	//	if err != nil {
+	//		return nil, err
+	//	}
+	//}
```

Comment:
For the things that we are commenting out, could we introduce more scenarios for test?
@nickzhums what do you think?

Reply:
Because ssh is required to create a linux virtual machine, I left the method.